apps/random/README.md

```diff
@@ -0,0 +1,134 @@
+# Random generator
+
+The TKey `Random generator` device application is an hardware based
+source of high quality random numbers. The generator is using a
+Hash_DRBG built around the BLAKE2s hash function. The application can
+also sign the random data in order to provide proof of its origin.
+
+## Client Go application
+
+The corresponding client application can be found in `cmd/runrandom`.
+
+## Application protocol
+
+`Random generator` has a simple protocol on top of the [TKey Framing
+Protocol](https://dev.tillitis.se/protocol/#framing-protocol) with the
+following requests:
+
+| *command*             | *FP length* | *code* | *data*                              | *response*            |
+|-----------------------|-------------|--------|-------------------------------------|-----------------------|
+| `CMD_GET_NAMEVERSION` | 1 B         | 0x01   | none                                | `RSP_GET_NAMEVERSION` |
+| `CMD_GET_RANDOM`      | 4 B         | 0x03   | Number of bytes, 1 < x < 126        | `RSP_GET_RANDOM`      |
```

Comment:
Why ask for just 126 bytes at a time? This leads to a lot of stuttering, basically Xmodem style. Another idea would be to ask for N bytes here, perhaps a u32, and then reply with many frames until N bytes (padded) has been sent back.

I'm OK with this at is is. Just wondering if it would lead to better performance. Yes, I think we wrote something about always doing ping-pong on the framing level... Nevertheless.

Reply:
Agreeing on the performance increase by removing the ping-pong. Will create an issue about this.